week_2/pet.rb

```diff
@@ -110,6 +118,15 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    ans = {}
+    for pet in pets
+      if ans.keys.include? pet.animal_type_id
```

Comment:
You can also just check if `ans[pet.animal_type_id]` is not nil

Reply:
I have made the changes